hypermodularization/what-is-hypermodular.md

```diff
@@ -0,0 +1,53 @@
+## What is "hypermodularization"?
+
+What I[^a] mean by this is a project (examples: [React](https://facebook.github.io/react/ 'Facebook Code: React'), [redux](http://redux.js.org/ 'redux')) that is explicitly designed with the expectation that a single dependency will **not** be self-contained. A developer-user (henceforth: user) will have to install several, perhaps even “many”[^b], dependencies in order to make full use of the project/framework or perhaps even to use it at all (as intended).
```

Comment:
I'm not sure this definition is great/clear, specifically this section:

> explicitly designed with the expectation that a single dependency will **not** be self-contained

Whilst I follow, after looking up the meaning of "self-contained", and understand the sentence. The phrase self-contained irks me in this context. Whilst you could **not** build a UI without `react` and the relevant render module - I would still argue `react` is self-contained in that it is a complete system in itself. 

I guess there could be more added here, and then maybe this definition would seem suitable. E.g. 

> What I mean by this is a project that is explicitly designed with the expectation that a single dependency will **not** be self-contained. A system that is 'hypermodularised' has been architected in such a way that there are logical separations of core and supporting code, to reduce it's weight when it is integrated.